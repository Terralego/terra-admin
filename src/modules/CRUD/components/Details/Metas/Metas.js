import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { Icon, Button } from '@blueprintjs/core';

import { generateURI } from '../../../config';

const Metas = ({ t, full, onSizeChange, match: { params: { layer } } }) => (
  <div className="CRUD-details__metas">
    <Button
      minimal
      icon={full ? 'minimize' : 'maximize'}
      onClick={onSizeChange}
    />
    <NavLink to={generateURI('layer', { layer })} title={t('CRUD.details.close')}>
      <span className="bp3-button bp3-minimal">
        <Icon icon="cross" />
      </span>
    </NavLink>
  </div>
);

Metas.propTypes = {
  full: PropTypes.bool,
  onSizeChange: PropTypes.func,
  match: PropTypes.shape({
    params: PropTypes.shape({
      layer: PropTypes.string,
    }),
  }),
  t: PropTypes.func,
};

Metas.defaultProps = {
  full: false,
  onSizeChange () {},
  match: {
    params: {
      layer: undefined,
    },
  },
  t: text => text,
};

export default Metas;
