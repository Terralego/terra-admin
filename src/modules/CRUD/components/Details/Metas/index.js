import { withRouter } from 'react-router-dom';
import { withNamespaces } from 'react-i18next';

import Metas from './Metas';
import compose from '../../../../../utils/compose';

export default compose(
  withRouter,
  withNamespaces(),
)(Metas);
