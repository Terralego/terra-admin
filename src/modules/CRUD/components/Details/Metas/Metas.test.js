import React from 'react';
import renderer from 'react-test-renderer';

import Metas from './Metas';

jest.mock('react-router-dom', () => ({
  NavLink: ({ children, ...props }) => <div {...props}>{children}</div>,
}));

jest.mock('@blueprintjs/core', () => ({
  Icon: () => <div>Icon close</div>,
  Button: ({ children, ...props }) => <button type="button" {...props}>{children}</button>,
}));


let props;
beforeEach(() => {
  props = {
    t: text => text,
    full: false,
    onSizeChange: jest.fn(),
    match: {
      params: { layer: 'fooLayer ' },
    },
  };
});

it('should render correctly with maximize icon', () => {
  const tree = renderer.create((
    <Metas {...props} />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render correctly with minimize icon', () => {
  const tree = renderer.create((
    <Metas {...props} full />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
