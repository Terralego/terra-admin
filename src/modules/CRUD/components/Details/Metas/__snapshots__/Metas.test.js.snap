// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with maximize icon 1`] = `
<div
  className="CRUD-details__metas"
>
  <button
    icon="maximize"
    minimal={true}
    onClick={[MockFunction]}
    type="button"
  />
  <div
    title="CRUD.details.close"
    to="/CRUD/map/fooLayer%20"
  >
    <span
      className="bp3-button bp3-minimal"
    >
      <div>
        Icon close
      </div>
    </span>
  </div>
</div>
`;

exports[`should render correctly with minimize icon 1`] = `
<div
  className="CRUD-details__metas"
>
  <button
    icon="minimize"
    minimal={true}
    onClick={[MockFunction]}
    type="button"
  />
  <div
    title="CRUD.details.close"
    to="/CRUD/map/fooLayer%20"
  >
    <span
      className="bp3-button bp3-minimal"
    >
      <div>
        Icon close
      </div>
    </span>
  </div>
</div>
`;
